### DISCUSSION TITLE
Resolve Bioscience data

### DISCUSSION THREAD

**Discussion by osmansharifi** [satijalab/seurat#5711] [Q&A]
Hello, 

I have a giant Seurat object containing snRNA-seq data from 10X and recently did spatial transcriptomics on the same samples using the resolve bioscience tech. I was wondering if Seurat has a method for integrating these two data sets similar to the data from vizgen or nanostring.

Thanks

**Comment by alikhuseynov** [comment]
Hey there, 
Did you find something on how to analyze RE data in R? 

Here is something our lab is developing currently, but not finished yet.
[active GitLab repo](https://gitlab.hzdr.de/michele.bortolomeazzi/RESOLVE_tools)
[archived GitHub repo](https://github.com/MicheleBortol/RESOLVE_tools)

Hope this helps.

**Reply by KaminskaLab** [reply]
Hi, is it possible to get access to the active repo?

**Reply by MicheleBortol** [reply]
Hi, sure. Sorry for the broken links. You should be able to access the repo here:
https://codebase.helmholtz.cloud/resolve_tools/resolve-analysis

**Reply by alikhuseynov** [reply]
@KaminskaLab you would need the [preprocessing part](https://codebase.helmholtz.cloud/resolve_tools/resolve-processing) first, before you go for Resolve analysis repo.

**Reply by KaminskaLab** [reply]
thank you a lot!

**Comment by marnixwkoops** [comment]
Hey, 

I am also trying to get my resolve data into a Seurat object, however I can not perform the preprocessing part. Can you show which files your function need to load it as a Seurat object. Or could you maybe upload an example dataset which was done using your preprocessing part? @MicheleBortol @alikhuseynov

**Reply by MicheleBortol** [reply]
Hi @marnixwkoops,

Sorry the documentation in our repo is a bit lacking. I'll describe the files here and then update the docs:
The `ReadResolve` function expects a folder with the following files:
- `*-cell_data.csv`:
 csv file with this header:
  ```
  cell,area,centroid.y,centroid.x,label,GENE1,GENE2,...
  ```
    - `cell` = table index, not used.
    - `centroid.y,centroid.x` = centroid coordinates (µm or pixel, see the `use.micron` argument)
    - `area` = area in pixel^2
    -  `GENE1,GENE2,...`= transcript counts per cell.
- `*-filtered_transcripts.txt` (either the raw output from resolve or the deduplicated output from [MindaGap](https://github.com/ViriatoII/MindaGap)):
    csv file with no header and these column:
    - x: pixels 
    - y: pixels
    - z: not used, but required
    - gene name
    - quality: not used, optional
 - `*-roi.zip` (optional, if not provided the centroids are used), zip format used by FiJi.
 - `*_mask.tiff`: segmentation mask from cellpose or similar tool (tiff file, used only to get the total width and height, so any image would work)
 - `*-gridfilled.tiff`: optional image to be added to the Seurat object for visualization. Generally we use the output from [MindaGap](https://github.com/ViriatoII/MindaGap), but any image would work. 

Sorry you had issues with the preprocessing, if you would like to open an issue with your error I can have a look. I haven't been working on the processing pipeline for a while as I was caught up in other projects. Now I would like to switch this tool to use the files produced with the MolKart processing pipeline for resolve https://github.com/nf-core/molkart/tree/dev which is honestly more up-to-date and better maintained than our processing tool, and then only maintain the Seurat loading part of the workflow.

**Reply by marnixwkoops** [reply]
Thanks a lot for your quick response! Does this mean that with a bit of shuffling the data around, you can directly load the raw data from resolve into Seurat, or is preprocessing always necessary?

**Reply by MicheleBortol** [reply]
For the transcripts and the DAPI image, no preprocessing is required (I would still recommend MindaGap though). However, you would always need to perform a segmentation and count the transcripts in each cell to generate the cell data which is then turned in the expression matrix which is the main input for Seurat. For these steps you can use any tool you like  (cellpose, mesmer, starfish,... for segmentation, https://github.com/nf-core/molkart/blob/dev/bin/spot2cell.py  or  https://codebase.helmholtz.cloud/resolve_tools/resolve-processing/-/blob/master/bin/extracter.py?ref_type=heads for counting the transcripts) and then just reformat the output to match the `ReadResolve` function.

**Reply by marnixwkoops** [reply]
Okay great. (BTW, resolve also gives a segmentation zip file. So, you only have to make the counts per cell file)

**Reply by Lucas-Maciel** [reply]
Dear @MicheleBortol, 
I ran molkart in my samples recently and now I will do the downstream analysis. I'm trying to adapt the obtained from spot2cell to your code, but it's not clear to me what is the column "label". What is it exactly? 

Thank you

**Reply by MicheleBortol** [reply]
Dear @Lucas-Maciel

The `label` is value of the pixels in the segmentation mask which correspond to the cell (with 0 assigned to the background). It should be fully equivalent to the `CellID` from the spot2cell output.

**Comment by MicheleBortol** [comment]
@marnixwkoops 
> Okay great. (BTW, resolve also gives a segmentation zip file. So, you only have to make the counts per cell file)
Cool, is this a new update from Resolve? Could you link an example?

**Comment by marnixwkoops** [comment]
[A1-roi.zip](https://github.com/satijalab/seurat/files/13602105/A1-roi.zip)

Yes, I have also some troubles linking the cellid to the names of the ROIs in the file that is created by Resolve, so that is something where a modification of the code is needed if you use the resolve segmentation directly. (BTW, this segmentation zip file they don't send to you directly but should be made inside the Resolve webpage)
